snap/types.go

```diff
@@ -187,10 +187,31 @@ const (
 type DaemonScope string
 
 const (
-	SystemDaemon DaemonScope = "system"
-	UserDaemon   DaemonScope = "user"
+	SystemDaemonScope        DaemonScope = "system"
+	UserDaemonScope          DaemonScope = "user"
+	GraphicalUserDaemonScope DaemonScope = "graphical-user"
 )
 
+// Returns if the daemon is a System daemon
+func (daemonScope DaemonScope) IsSystemDaemon() bool {
+	switch daemonScope {
+	case SystemDaemonScope:
+		return true
+	default:
```

Comment:
Don't remeber now but there's no warning for unhandled cases in go (when using iota enums right)? As otherwise it would have been nicer to be explicit